makes buster arm easier to use by turning it into a martial art

LazennG opened the pull request:
# Document the changes in your pull request

- wire snatch on an anchored object no longer makes you risk a wound on the way there
- damage reduction moved from 65% to 50%
- buster arm makes the user insulated now
- files for slam/grapple/mop were deleted since they were replaced in the martial art
- slam damage reduced to 20 and wire snatch immobilize reduced to 1 second
-slam cooldown is now 0.8 seconds like normal melee attacks


Buster arm is a martial art now
- help intenting yourself puts the grappling wire in your current hand
- disarm intenting and clicking somewhere makes you mop in that cardinal direction
- harm intenting someone will slam them instead
- grab intenting something picks it up and makes your next click with grab intent throw it towards the destination
  -grabbed mobs can be held for up to 10 seconds but can resist their way out of it after the first second of being grabbed
  -grabbed objects can be held until they're thrown but will be dropped if the user tries throwing them while the user is unable to do it (cuffed/horizontal/etc)
- megabuster is still a button



https://user-images.githubusercontent.com/58535870/224519374-7a7b6686-3429-4e9c-be37-f4c9cf8e43ae.mp4



# Wiki Documentation

https://wiki.yogstation.net/wiki/Syndicate_Items
Since it's a martial art i'll try describing it like one:
 replace 

"A box containing a combat-focused prosthetic left arm that can be attached on contact; It is intended for close combat and possesses immense strength. With it, the user can send people and heavy objects flying and even tear down solid objects like they're wet paper. To close the distance with ranged opponents, a grappling hook can be ejected from the arm which momentarily keeps victims in place. "

 with
 
A powerful arm that grants the user the strength to use a brutish martial art.
(under a "recall teachings" dropdown)
Wire snatch: Help intent on yourself. Get a wire that can fire 3 times and immobilize people for 1 second
Mop the floor: Disarm intent. Rush in the general direction of the target, dealing considerable damage to people in the way and whatever you ram them into.
Grapple: Grab. Pick up the unanchored target and target a location while on grab intent to throw it there. Things hit with the thrown object will make all objects involved take damage and unanchored objects hit will move with the original projectile. People can be held for up to 10 seconds but can escape by resisting. Mechs and vending machines can be picked up as well.
Slam: Harm. The target will be moved to the space behind you, taking bonus damage if they collide with something solid. Objects they hit will take heavy damage and walls behind the user will be destroyed.
Megabuster: Deal heavy damage to the target person or object and send unanchored ones flying similar to grapple. A direct hit will break reinforced walls. Targets sent flying with this attack can be sent through walls and windows for bonus damage.

also change the 65 to 50 for damage reduction
# Changelog

:cl:  @LazennG  @thegoldencat413 @slicerv @solwashere @milk-is-cool @SapphicOverload 
rscadd: Added new martial art that's just buster arm moves  
rscdel: Removed old buster arm move files that got replaced
bugfix: grapple can no longer work on containers you're inside of  
tweak: tweaked the buster arm to rely less on action buttons
/:cl:
thanks to the people above for helping test

LazennG committed:
how long until a mistake is found

LazennG committed:
Merge remote-tracking branch 'upstream/master' into buster-art

LazennG committed:
test

LazennG committed:
C

LazennG committed:
Update code/datums/martial/buster_style.dm

Co-authored-by: tattax <71668564+tattax@users.noreply.github.com>

LazennG committed:
Update code/datums/martial/buster_style.dm

Co-authored-by: tattax <71668564+tattax@users.noreply.github.com>

LazennG committed:
Update code/datums/martial/buster_style.dm

Co-authored-by: tattax <71668564+tattax@users.noreply.github.com>

LazennG committed:
Merge branch 'buster-art' of https://github.com/LazennG/bad-idea-box into buster-art

LazennG committed:
Update code/datums/martial/buster_style.dm

Co-authored-by: tattax <71668564+tattax@users.noreply.github.com>

LazennG committed:
Update code/datums/martial/buster_style.dm

Co-authored-by: tattax <71668564+tattax@users.noreply.github.com>

LazennG committed:
Merge branch 'buster-art' of https://github.com/LazennG/bad-idea-box into buster-art

LazennG committed:
Merge branch 'buster-art' of https://github.com/LazennG/bad-idea-box into buster-art

LazennG committed:
Update code/datums/martial/buster_style.dm

Co-authored-by: tattax <71668564+tattax@users.noreply.github.com>

LazennG committed:
Update buster_style.dm

LazennG committed:
Merge branch 'buster-art' of https://github.com/LazennG/bad-idea-box into buster-art

LazennG committed:
naming is too distracting

LazennG committed:
what the fuck is examine_block and /n/

LazennG committed:
Update code/game/objects/items/devices/busterarm/megabuster.dm

Co-authored-by: Redd <36747151+ReddicusDragon@users.noreply.github.com>

ynot01 reviewed (commented):
> disarm intenting and clicking somewhere makes you mop in that cardinal direction

Would vastly prefer if this just took you in the direction where you clicked

This can be achieved by using `get_step_towards` instead of `get_step`

LazennG committed:
arg

LazennG committed:
cd

ynot01 reviewed (changes requested):
Runtime when bed is deleted after 10 seconds
```
[02:12:56] Runtime in garbage.dm, line 395: /obj/structure/bed/grip destroy proc was called multiple times, likely due to a qdel loop in the Destroy logic
proc name: qdel (/proc/qdel)
usr: Ynot01/(Raegan Adams)
usr.loc: (Starboard (E) Primary Hallway (141, 139, 2))
src: null
call stack:
qdel(the buster arm (/obj/structure/bed/grip), 0)
the buster arm (/obj/structure/bed/grip): post unbuckle mob(the monkey (237) (/mob/living/carbon/monkey))
the buster arm (/obj/structure/bed/grip): unbuckle mob(the monkey (237) (/mob/living/carbon/monkey), 1)
the buster arm (/obj/structure/bed/grip): unbuckle all mobs(1)
the buster arm (/obj/structure/bed/grip): Destroy(0)
the buster arm (/obj/structure/bed/grip): Destroy(0)
the buster arm (/obj/structure/bed/grip): Destroy(0)
the buster arm (/obj/structure/bed/grip): Destroy(0)
qdel(the buster arm (/obj/structure/bed/grip), 0)
/datum/callback (/datum/callback): Invoke()
world: PushUsr(Raegan Adams (/mob/living/carbon/human), /datum/callback (/datum/callback))
/datum/callback (/datum/callback): InvokeAsync()
Timer (/datum/controller/subsystem/timer): fire(0)
Timer (/datum/controller/subsystem/timer): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0)
```

ynot01 reviewed (changes requested):
When throwing a mob and they fail to travel any distance (ie a wall or dense object in front of you), they will remain attached to you until the 10 seconds is up and you will not be able to throw them again

ynot01 reviewed (changes requested):
When shift clicking (to examine) or alt clicking (to tile search) or even clicking on an inventory slot, the buster arm will assume you are trying to perform a move and will trigger a mop or grab or what have you

ynot01 reviewed (changes requested):
Despite someone being removed from your grapple, you will still "store" them as your last grabbed, unthrown victim, and will be able to throw them from that point onwards at any time as long as you don't grab-click until then and are adjacent to them.

This can be abused intentionally by using a slam to "dislodge" someone out of your grapple and then throw them at any later point as long as you are adjacent to them

ynot01 reviewed (changes requested):
Picking up a vendor opens its menu, which is very disruptive

LazennG committed:
i

-bed no longer deletes itself in 10 seconds cause people can just resist to get out anyway
-throwing a target 0 meters and throwing while you shouldn't be able to should properly separate them from you now
-cant throw people who arent grappled anymore
-will work on the vending machine thing and probably grapple suplex lob thing tomorrow

LazennG committed:
ignore that error

LazennG committed:
Revert "ignore that error"

This reverts commit 9ad71c1eec01548441cf5eb6701507d9f28e0650.

LazennG committed:
Revert "i"

This reverts commit e387df4fa1f6e0ef3f8c6513fecf751ccadccf03.

LazennG committed:
Revert "Revert "i""

This reverts commit aaa9205ba49838a56d6ff4e179513e3a422d7dd3.

LazennG committed:
fixed the slam thing

LazennG committed:
vending machine menus no longer open on picking htem up

LazennG committed:
you can examine now

LazennG committed:
changes damage reduction to 50% from 65%

ynot01 reviewed (changes requested):
> When throwing a mob and they fail to travel any distance (ie a wall or dense object in front of you), they will remain attached to you until the 10 seconds is up and you will not be able to throw them again

Still an issue, except now it seems that the bed never auto-deletes itself

Also, when mopping someone into crit, they will sometimes have their angles broken and appear standing or upside-down

LazennG committed:
OBLITERATES beds

LazennG committed:
STOP BREAKING MY LEGS PLEASE THANKS

-wire snatch now just forcemoves you to the destination instead of making you risk a fucking hairline fracture by crashing into a person or the wall

LazennG committed:
fixed animations most likely
